cpp/src/toppra/algorithm.cpp

```diff
@@ -105,12 +105,28 @@ ReturnCode PathParametrizationAlgorithm::computeFeasibleSets() {
   return ret;
 }
 
+void PathParametrizationAlgorithm::setGridpoints(const Vector& gridpoints)
+{
+  Bound I (m_path->pathInterval());
+  auto s = gridpoints.size();
+  if (s != m_N + 1)
+    /// \todo shouldn't we set m_N instead ?
```

Comment:
Yes indeed there is no reason to fix m_N if we are setting the gridpoints directly.